Crash in find enclosing parentheses

## Lapce Version

0.2.4 (Current HEAD)

## System information

Linux

## Describe the bug
When opening a new markdown file and typing, the editor crashed.  
The log:
```rust
[2022-11-22][20:24:24][panic][ERROR] thread 'main' panicked at 'index out of bounds: the len is 0 but the index is 0': /home/minus/.cargo/registry/src/github.com-1ecc6299db9ec823/lapce-xi-rope-0.3.1/src/rope.rs:534
(cut)
   8: lapce_xi_rope::rope::<impl lapce_xi_rope::tree::Node<lapce_xi_rope::rope::RopeInfo>>::byte_at
   9: lapce_core::syntax::Syntax::find_enclosing_parentheses
  10: lapce_data::editor::LapceEditorBufferData::update_signature
  11: lapce_data::editor::LapceEditorBufferData::apply_deltas
  12: <lapce_data::editor::LapceEditorBufferData as lapce_data::keypress::KeyPressFocus>::run_command
```

--- a/lapce-core/src/syntax/mod.rs
+++ b/lapce-core/src/syntax/mod.rs
@@ -792,6 +792,10 @@ impl Syntax {
     ) -> Option<(usize, usize)> {
         let tree = self.layers.try_tree()?;
         let mut node = tree.root_node().descendant_for_byte_range(offset, offset)?;
+        // If there is no text then the document can't have any bytes
+        if self.text.is_empty() {
+            return None;
+        }
 
         loop {
             let start = node.start_byte();
@@ -815,6 +819,10 @@ impl Syntax {
     pub fn find_enclosing_pair(&self, offset: usize) -> Option<(usize, usize)> {
         let tree = self.layers.try_tree()?;
         let mut node = tree.root_node().descendant_for_byte_range(offset, offset)?;
+        // If there is no text then the document can't have any bytes
+        if self.text.is_empty() {
+            return None;
+        }
 
         loop {
             let start = node.start_byte();

